src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -527,11 +527,221 @@ private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles
                     {
                         if (!parseDecimal)
                         {
-                            number.scale = 0;
+                            integer.scale = 0;
                         }
                         if ((state & StateDecimal) == 0)
                         {
-                            number.sign = false;
+                            integer.sign = false;
+                        }
+                    }
+                    str = p;
+                    return true;
+                }
+            }
+            str = p;
+            return false;
+        }
+
+        private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles styles, ref FloatingPointBuffer floatingPoint, NumberFormatInfo info, bool parseDecimal)
```

Comment:
This is the other big "duplicated" code, but it will likely need to change a bit between the two implementations. I'll also give it some more thought, to see if we can share some of the code here.